packages/gatsby-theme-docs/src/components/code-example.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { CodeBlock, ContentNotifications } from '@commercetools-docs/ui-kit';
+import useCodeExamples from '../hooks/use-code-examples';
+
+function CodeExample(props) {
+  const codeExamples = useCodeExamples();
+
+  const codeExample = codeExamples.find(example => {
+    return example.absolutePath.includes(props.file);
+  });
+
+  if (!codeExample) {
+    return reportError('Code example does not exist');
+  }
+
+  const language = props.file.split('.').pop();
+
+  return (
+    <CodeBlock
+      content={codeExample.content}
+      language={language}
+      title={props.title}
+      highlightLines={props.highlightLines}
+      noPromptLines={props.noPromptLines}
+    />
+  );
+}
+
+function reportError(errorMsg) {
+  if (process.env.NODE_ENV !== 'production') {
+    return <ContentNotifications.Error>{errorMsg}</ContentNotifications.Error>;
+  }
+
+  throw new Error(errorMsg);
+}
+
+CodeExample.propTypes = {
+  file: PropTypes.string.isRequired,
```

Comment:
Isn't `path` a more appropriate name to use? I find `file` a bit confusing, as it's not clear it's a string path.

```
<CodeExample file="foo.js" />

<CodeExample path="foo.js" />
```